language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -692,6 +693,20 @@ private static String generateVariableName(int suffix, String name, Set<String>
         return newName;
     }
 
+    /**
+     * Checks if the provided identifier is valid as per the ballerina specification.
+     *
+     * @param identifier Identifier to be checked for validity
+     * @return True, if the identifier is valid as per the ballerina specification
+     */
+    public static boolean isIdentifierValid(String identifier) {
```

Comment:
Maybe we can change the name to `isValidIdentifier`

Reply:
Done!